python/hail/utils/tutorial.py

```diff
@@ -0,0 +1,164 @@
+
+        tmp_annot = os.path.join(tmp_dir, '1kg_annotations.txt')
+        source = resources['1kg_annotations']
+        info(f'downloading 1KG annotations ...\n'
+             f'  Source: {source}')
+        urlretrieve(source, tmp_annot)
+        hl.hadoop_copy(local_path_uri(tmp_annot), annotations_path)
+        info('Done!')
+    else:
+        info('1KG files found!')
+
+
+def get_movie_lens(output_dir, overwrite: bool = False):
+    """Download public Movie Lens dataset.
+
+    Notes
+    -----
+    See the
+    `MovieLens website <https://grouplens.org/datasets/movielens/100k/>`__
+    for more information about this dataset.
+
+    Parameters
+    ----------
+    output_dir
+        Directory in which to write data.
+    overwrite
+        If ``True``, overwrite existing files/directories at those locations.
+    """
+
+    jhc = Env.hc()._jhc
+
+    _mkdir(jhc, output_dir)
+
+    paths = [os.path.join(output_dir, x) for x in ['movies.ht', 'ratings.ht', 'users.ht']]
+    if overwrite or any(not Env.jutils().dirExists(jhc, f) for f in paths):
+        init_temp_dir()
+        source = resources['movie_lens_100k']
+        tmp_path = os.path.join(tmp_dir, 'ml-100k.zip')
+        info(f'downloading MovieLens-100k data ...\n'
+             f'  Source: {source}')
+        urlretrieve(source, tmp_path)
+        with zipfile.ZipFile(tmp_path, 'r') as z:
+            z.extractall(tmp_dir)
+
+        user_table_path = os.path.join(os.path.join(tmp_dir, 'ml-100k', 'u.user'))
+        movie_table_path = os.path.join(os.path.join(tmp_dir, 'ml-100k', 'u.item'))
+        ratings_table_path = os.path.join(os.path.join(tmp_dir, 'ml-100k', 'u.data'))
+        assert (os.path.exists(user_table_path))
+        assert (os.path.exists(movie_table_path))
+        assert (os.path.exists(ratings_table_path))
+
+        user_cluster_readable = Env.jutils().copyToTmp(jhc, local_path_uri(user_table_path))
+        movie_cluster_readable = Env.jutils().copyToTmp(jhc, local_path_uri(movie_table_path))
+        ratings_cluster_readable = Env.jutils().copyToTmp(jhc, local_path_uri(ratings_table_path))
+
+        [movies_path, ratings_path, users_path] = paths
+
+        genres = ['Action', 'Adventure', 'Animation',
+                  "Children's", 'Comedy', 'Crime',
+                  'Documentary', 'Drama', 'Fantasy',
+                  'Film-Noir', 'Horror', 'Musical',
+                  'Mystery', 'Romance', 'Sci-Fi',
+                  'Thriller', 'War', 'Western']
+
+        # utility functions for importing movies
+        def field_to_array(ds, field):
+            return hl.cond(ds[field] != 0, hl.array([field]), hl.empty_array(hl.tstr))
+
+
+        def fields_to_array(ds, fields):
+            return hl.flatten(hl.array([field_to_array(ds, f) for f in fields]))
+
+        def rename_columns(ht, new_names):
+            return ht.rename({k: v for k, v in zip(ht.row, new_names)})
+
+        info(f'importing users and writing to {users_path} ...')
```

Comment:
I think users, movies, and ratings should be in some form of quotes in these three messages